tests/general/RegexLintBearTest.py

```diff
@@ -0,0 +1,84 @@
+from os.path import abspath
+from queue import Queue
+
+from bears.general.RegexLintBear import RegexLintBear
+from coalib.testing.LocalBearTestHelper import (
+    LocalBearTestHelper, execute_bear)
+from coalib.results.Result import Result
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+
+
+test_good_python_file = """
+some_regex = r'[a-zA-Z]]'
+"""
+
+test_bad_python_file = """
+some_regex = r'(else|elseif)'
+"""
+
+test_good_cpp_file = """
+char some_regex[] = "[a-zA-Z]]";
+"""
+
+test_bad_cpp_file = """
+char some_regex[13] = "(else|elseif)";
+"""
+
+test_re_error_file = """
+some_regex = r'*ab' # This should be skipped
+some_other_regex = r'[a-z]'
+"""
+
+BAD_MESSAGE = ('E105:argv:root:0: Potential out of order alternation between '
+               '\'else\' and \'elseif\'\n'
+               '  \'(else|elseif)\'\n'
+               '         ^ here\n')
+
+
+class RegexLintBearTest(LocalBearTestHelper):
+
+    def setUp(self):
+        self.section = Section('')
+        self.queue = Queue()
+        self.uut = RegexLintBear(self.section, self.queue)
+
+    def test_good_python_file(self):
+        self.section.append(Setting('language', 'python 3'))
+        with execute_bear(self.uut, abspath('good_python_file'),
+                          test_good_python_file.splitlines(True)) as results:
+            self.assertEqual(results, [])
+
+    def test_bad_python_file(self):
+        self.section.append(Setting('language', 'python 3'))
+        with execute_bear(self.uut, abspath('bad_python_file'),
+                          test_bad_python_file.splitlines()) as results:
+            self.assertEqual(results[0],
+                             Result.from_values(origin=self.uut,
+                                                message=BAD_MESSAGE,
+                                                file='bad_python_file',
+                                                line=2, column=15,
+                                                end_line=2, end_column=29))
+
+    def test_good_cpp_file(self):
+        self.section.append(Setting('language', 'cpp'))
+        with execute_bear(self.uut, abspath('good_cpp_file'),
+                          test_good_cpp_file.splitlines()) as results:
+            self.assertEqual(results, [])
```

Comment:
This has no line number.  Use our pre-rolled helpers for more maintainable code.